alyx/actions/admin.py

```diff
@@ -104,6 +104,26 @@ def queryset(self, request, queryset):
             return queryset.all()
 
 
+class HasNarrativeFilter(DefaultListFilter):
+    title = 'narrative'
+    parameter_name = 'narrative'
+
+    def lookups(self, request, model_admin):
+        return (
+            (None, 'All'),
+            ('narrative', 'Narrative'),
+            ('no_narrative', 'No narrative'),
+        )
+
+    def queryset(self, request, queryset):
+        if self.value() == 'narrative':
+            return queryset.filter().exclude(narrative__in=['', 'auto-generated session'])
```

Comment:
no need for `.filter()` here. As we use PostgreSQL as a backend we could even exclude whitespace only narratives using regex.